src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/split_fc.cpp

```diff
@@ -0,0 +1,213 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/constant_folding.hpp"
+#include <transformations/utils/utils.hpp>
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/variadic_split.hpp"
+#include "transformations/cpu_opset/common/op/fully_connected.hpp"
+
+#include "split_fc.hpp"
+
+#include "itt.hpp"
+
+ov::intel_cpu::SplitFC::SplitFC(int sub_stream_num) {
+    MATCHER_SCOPE(SplitFC);
+    auto fc_m = ov::pass::pattern::wrap_type<ov::intel_cpu::FullyConnectedNode>();
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        const auto& fc_node = pattern_map.at(fc_m).get_node_shared_ptr();
+        auto& rt_info = fc_node->get_rt_info();
+        if (rt_info.count("parallelDomain")) {
```

Comment:
Not critical: It is better to create a class for this `rt_info` instead of just `std::string` usage. This will allow us to override `ov::RuntimeAttribute` methods:

1. `bool is_copyable()`: if I understand correctly, this rt info can be set only in this transformation and only for the split `FullyConnected` nodes, so this method must always return false because we need to avoid this info propagation to other nodes
2. `std::string to_string()`: this will help us to get more info in case of debug serialization

[This plugin rt info](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/utils/rt_info/memory_formats_attribute.hpp) can be used as an example.

Reply:
> Not critical: It is better to create a class for this `rt_info` instead of just `std::string` usage. This will allow us to override `ov::RuntimeAttribute` methods:
> 
> 1. `bool is_copyable()`: if I understand correctly, this rt info can be set only in this transformation and only for the split `FullyConnected` nodes, so this method must always return false because we need to avoid this info propagation to other nodes
> 2. `std::string to_string()`: this will help us to get more info in case of debug serialization
> 
> [This plugin rt info](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/utils/rt_info/memory_formats_attribute.hpp) can be used as an example.

I see. Your advice really makes sense. In order to better debug, this attribute should includes the info of op split path and then dump by `to_string` api. At this milestone, it may not be possible to implement and test this feature in time. I plan to implement it in the follow-up PR.